### ISSUE TITLE
RabbitMQ Plugins

### ISSUE THREAD

**Issue by occasl** [rancher/community-catalog#414]
I would like to install the [Federation Plugin](https://www.rabbitmq.com/federation.html) after installing RabbitMQ from the community catalog. How can I accomplish this so that it stays persistent after any restarts of the environment?

**Comment by occasl** [comment]
It looks like the plugin using a volume to persist to disk so when I SSH into the docker image and install the plugins per the usual process, they stay there upon reboot.

**Comment by ddmng** [comment]
@occasl 
If you scale up/down the number of nodes or simply the containers are relocated by Rancher you'll loose the plugin and Federation could stop working. I think the best way is to add the plugin in RabbitMQ Rancher Stack so that you'll always have it in case you need it.

I can't work on it right now but if you could, this is the -conf image that is configured in Rancher:
https://hub.docker.com/r/rdaneel/rabbitmq-conf/
and the sources are here:
https://github.com/ddmng/catalog-dockerfiles/tree/master/rabbitmq/0.2.0

**Comment by occasl** [comment]
Yes, good point. Alright let me take a look and I'll send a pull request. Is there a way I could point my rancher instance to my fork to test it?

**Comment by ddmng** [comment]
Just add an additional catalog to rancher pointing to a forked community-catalog version. For rabbitmq-conf image, you could either use an image on your docker hub account or use a local one and then make the catalog point to a new version on mine before merging.
